src/chainparamsbase.cpp

```diff
@@ -17,10 +17,12 @@ const std::string CBaseChainParams::REGTEST = "regtest";
 
 void SetupChainParamsBaseOptions()
 {
+    gArgs.AddArg("-chain=<chain>", "Use the chain <chain> (default: main). Reserved values: main, test, regtest", false, OptionsCategory::CHAINPARAMS);
```

Comment:
I am wondering if this option should be hidden like regtest...

Effectively this reverts the option sanitization check at initialization, where typos would be rejected for e.g. `-tesnet`. Now, `-chain=tesnet` is neither an error, nor does sync with testnet3.

Reply:
Ah, sorry. I mean `-chain=tes` (with a typo) would no longer throw an error nor sync with testnet. I just wanted to point out that behaviour change.

Reply:
maybe it could log/print an extra message like "using custom chain xyz" when `chain` is not equal to `main`, `test` or `regtest`? At least that would make the potential typo easier to spot

Reply:
@flack That seems simple and reasonable. @MarcoFalke would that solve your concern?